lib/src/controller/controller.dart

```diff
@@ -575,4 +576,20 @@ class AppController with ChangeNotifier {
         break;
     }
   }
+
+  Future<void> markAsRead(int postId, bool read) async {
+    if (serverSoftware == ServerSoftware.lemmy && isLoggedIn) {
+      api.threads.markAsRead(postId, read);
+    }
+    if (await isRead(postId)) return;
+    await _readStore.add(db, {'postId': postId, 'read': read, 'account': _selectedAccount});
+  }
+
+  Future<bool> isRead(int postId) async {
+    final records = (await _readStore.find(db, finder: Finder(
+      filter: Filter.equals('postId', postId)
+    )));
+
+    return records.where((record) => record.value['account'] == _selectedAccount).firstOrNull != null;
```

Comment:
Is it possible to merge/cleanup these two checks together, like change the db query to not just filter by `postId`, but to also filter by `account` at the same time?